### PR TITLE
Add buffer size configuration for reportinglib

### PR DESCRIPTION
**Pull request by jorblancoa**
- Read buffer size from neurodamus report.conf file

### CODE REVIEW

**Comment by pramodk** [comment]
@jorblancoa  : looks good. one related thing : there is an existing command line option to change number of buffered steps:

```
$coreneuron_exec -h

USAGE:

OPTIONS:

--num-report-mindelay ARG   Number of min delay intervals to be buffered during
                            reporting (int).
```

can you change this option to `report-buffer-size` (default 4MB) ? That should allow to change size of buffer from command line. See coreneuron/coreneuron/nrniv/nrnoptarg.cpp.

File: coreneuron/utils/reports/nrnreport.cpp [hunk 1]
```diff
@@ -241,7 +241,8 @@ void register_soma_report(NrnThread& nt,
                            config.report_dt, sizemapping, (char*)config.type_str, extramapping,
                            (char*)config.unit);
 
-        /** add extra mapping */
+        records_set_report_max_buffer_size_hint ((char*)config.output_path, config.buffer_size);
```
Comment (jorblancoa) [inline coreneuron/utils/reports/nrnreport.cpp]: Should this option `report-buffer-size` overwrite the `records_set_report_max_buffer_size_hint` that reads the buffer size from the report.conf file?

File: coreneuron/utils/reports/nrnreport.cpp [hunk 2]
```diff
@@ -376,20 +374,18 @@ void nrn_flush_reports(double t) {
...
 
-// number of min delay intervals that will be buffered with reports
-// @todo : this will be replaced by buffer size in MB
-void set_num_mindelay_to_buffer(int n) {
-    num_min_delay_to_buffer = n;
+// Size in MB of the report buffers
+void set_report_buffer_size(int n) {
+    size_report_buffer = n;
+#ifdef ENABLE_REPORTING
+        records_set_max_buffer_size_hint(size_report_buffer);
```
Comment (ferdonline) [inline coreneuron/utils/reports/nrnreport.cpp]:
```suggestion
    records_set_max_buffer_size_hint(size_report_buffer);
```

File: coreneuron/utils/reports/report_configuration_parser.cpp [hunk 3]
```diff
@@ -79,10 +79,11 @@ std::vector<ReportConfiguration> create_report_configurations(const char* conf_f
         ReportConfiguration& report = reports[reports.size() - 1];
         // mechansim id registered in coreneuron
         report.mech_id = -1;
-        fgets(raw_line, MAX_FILEPATH_LEN, fp);
-        sscanf(raw_line, "\n%s %s %s %s %s %s %d %lf %lf %lf %d\n", report.name, report.target_name,
+        report.buffer_size = 4; //default size to 4 Mb
+	fgets(raw_line, MAX_FILEPATH_LEN, fp);
+	sscanf(raw_line, "\n%s %s %s %s %s %s %d %lf %lf %lf %d %d\n", report.name, report.target_name,
```
Comment (ferdonline) [inline coreneuron/utils/reports/report_configuration_parser.cpp]:
```suggestion
        sscanf(raw_line, "\n%s %s %s %s %s %s %d %lf %lf %lf %d %d\n", report.name, report.target_name,
```

Comment (ferdonline) [reply coreneuron/utils/reports/report_configuration_parser.cpp]: A couple of lines here got indented with tabs...

**Pull request merged**